billing/signals.py

```
@@ -0,0 +1,89 @@
+            )
+
+
+@receiver(post_save, sender=Transaction)
+def update_wallet_after_transaction(sender, instance, created, **kwargs):
+    """
+    Update wallet balance after successful transaction.
+    This is for external payment gateway transactions.
+    """
+    if not created:
+        # Only process on update, not on create
+        if instance.status == 'completed' and instance.transaction_type in ['credit', 'refund']:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)